cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -66,23 +71,14 @@ cdef class Buffer:
     @classmethod
     def _init(cls, ptr: DevicePointerT, size_t size, mr: MemoryResource | None = None):
         cdef Buffer self = Buffer.__new__(cls)
-        self._ptr = <uintptr_t>(int(ptr))
```

Comment:
This looks incorrect to me.

- uintptr_t vs intptr_t - intptr_t has a reduced range of representative pointer value due to the signed bit
- int(ptr) - explicitly narrowing conversion to a 32-bit python int before promoting to a 64-bit data type. We are throwing away data in this conversion.

- uintptr_t  =           0 to 2^64 − 1
- intptr_t     = −2^63 to 2^63 − 1